pkg/process/checks/discovery_check.go

```diff
@@ -0,0 +1,101 @@
+package checks
+
+import (
+	"fmt"
+
+	model "github.com/DataDog/agent-payload/process"
+	"github.com/DataDog/datadog-agent/pkg/process/config"
+	"github.com/DataDog/datadog-agent/pkg/process/procutil"
+)
+
+// ProcessDiscovery is ProcessDiscoveryCheck singleton. ProcessDiscoveryCheck should not be instantiated elsewhere.
+var ProcessDiscovery = &ProcessDiscoveryCheck{}
+
+// ProcessDiscoveryCheck is a check that gathers basic process metadata and sends it to the process discovery service.
+// It uses its own ProcessDiscovery payload, which is intended to be read by the process_discovery kafka topic.
+// The goal of this check is to collect information about possible integrations that may be enabled by the end user.
+type ProcessDiscoveryCheck struct {
+	probe      *procutil.Probe
+	info       *model.SystemInfo
+	initCalled bool
+}
+
+// Init initializes the ProcessDiscoveryCheck. It is a runtime error to call Run without first having called Init.
+func (d *ProcessDiscoveryCheck) Init(_ *config.AgentConfig, info *model.SystemInfo) {
+	d.probe = procutil.NewProcessProbe()
+	d.info = info
+	d.initCalled = true
+}
+
+// Name returns the name of the ProcessDiscoveryCheck. In this case it is "process_discovery"
+func (d *ProcessDiscoveryCheck) Name() string { return config.DiscoveryCheckName }
+
+// RealTime returns a value that says whether this check should be run in real time.
+func (d *ProcessDiscoveryCheck) RealTime() bool { return false }
+
+// Run collects process metadata, and packages it into a CollectorProcessDiscovery payload to be sent to process-intake.
+// It is a runtime error to call Run without first having called Init.
+func (d *ProcessDiscoveryCheck) Run(cfg *config.AgentConfig, groupID int32) ([]model.MessageBody, error) {
+	if !d.initCalled {
+		return nil, fmt.Errorf("ProcessDiscoveryCheck.Run called before Init")
+	}
```

Comment:
To be perfectly honest, I really wish the other checks also threw this exception! Currently, if you don't call Init, the expected behavior of the other checks is a segfault when you try to access the check's members 😰 Let me know what you think though, it may be better to remove it for consistency sake.

Instantiating the check on import unfortunately won't solve the problem completely, since we still use data from `info` which can only retrieved through the Init function. With that said it definitely mitigates the problem quite significantly, so I will go ahead and make that change!